### PR TITLE
Add compute_wasv in gmfs/doppler

### PR DESCRIPTION
**Pull request by ACHMartin**
Creation of the Wrap-up compute_wasv function 
Modification of mouche12 to include sub-function within mouche12
Creation of test_seastar_doppler_package_adrien_v0 notebook to test the basic functionality of compute_wasv.

### CODE REVIEW

File: seastar/gmfs/doppler.py [hunk 1]
```diff
@@ -4,21 +4,109 @@
...
 from scipy.constants import c  # speed of light in vacuum
 
 
-# import pdb
+def compute_wasv(ds_env, gmf, **kwargs):
+    """
+    Compute the Wind-wave Artefact Surface Velocity (WASV)
+    as function of geophysical conditions defined in 'ds_env'.
+    :param ds_env: A Dataset containing WindSpeed, WindDirection,
+    IncidenceAngleImage, LookDirection, Polarization
```
Comment (DavidMcCann-NOC) [inline seastar/gmfs/doppler.py:17]: We will need to either pass separate iterations of ds_env with different LookDirection variables (for fore/aft/mid antennas) or make use of a dimension (Antenna) in the LookDirection array. The latter is a little beyond me at this stage, so perhaps its better if I pass in fore/aft specific ds_env datasets for the time being

File: seastar/gmfs/doppler.py [hunk 2]
```diff
@@ -4,21 +4,109 @@
...
+    # Initialisation
+    central_wavelength = wavenumber2wavelength(ds_env.CentralWavenumber)
+
+    relative_wind_direction = \
+        np.abs(
+            np.mod(
+                ds_env.WindDirection - ds_env.LookDirection + 180,
+                360
+            ) - 180
+        )  # This function propably already exist TODO to check
```
Comment (DavidMcCann-NOC) [inline seastar/gmfs/doppler.py:33]: This does already exist, although I haven't merged it with main yet. I'll edit this and link it to the function in .tools

File: seastar/gmfs/doppler.py [hunk 3]
```diff
@@ -4,21 +4,109 @@
...
+
+    ds_wa = xr.Dataset()
+    ds_wa['WASV'] = (ds_env.dims, wasv_rsv)
+    return ds_wa
+
+
+# def convert_doppler_shift_freq_to_radial_surface_velocity()
+def wavenumber2wavelength(wavenumber):
+    wavelength = 2 * np.pi / wavenumber
+    return wavelength
```
Comment (DavidMcCann-NOC) [inline seastar/gmfs/doppler.py:83]: This seems like a good fit for the .tools subpackage, I can see us calling this in future functions as well

File: seastar/gmfs/doppler.py [hunk 4]
```diff
@@ -4,21 +4,109 @@
...
+import pdb
 from scipy.constants import c  # speed of light in vacuum
 
 
-# import pdb
+def compute_wasv(ds_env, gmf, **kwargs):
+    """
+    Compute the Wind-wave Artefact Surface Velocity (WASV)
+    as function of geophysical conditions defined in 'ds_env'.
+    :param ds_env: A Dataset containing WindSpeed, WindDirection,
```
Comment (DavidMcCann-NOC) [inline seastar/gmfs/doppler.py:16]: We will either need to pass in look-direction specific ds_env datasets (which will mean multiple instances of Windspeed/Direction etc across multiple datasets) or have an 'Antenna' dimension to the LookDirection variable. Not sure exactly how to implement this as I'm still learning xarray's specifics but it seems to me to be the better way to go.

**Review by DavidMcCann-NOC — COMMENTED** [review]
Comments added, but the main thing is for me to organise the input datasets. Otherwise everything should be easily mergeable with my code.

I'll make some changes on my end and once its all working i'll merge my branch with main and create a 'backup' versioned branch.

File: seastar/gmfs/doppler.py (hunk 4 above)
Comment (DavidMcCann-NOC) [reply seastar/gmfs/doppler.py:16]: Oops, sorry - duplicated comment!

File: seastar/gmfs/doppler.py (hunk 2 above)
Comment (ACHMartin) [reply seastar/gmfs/doppler.py:33]: perfect, please let me know when it is done, so we can close this point. Thanks

File: seastar/gmfs/doppler.py (hunk 1 above)
Comment (ACHMartin) [reply seastar/gmfs/doppler.py:17]: I will update the notebook to take this into account

File: seastar/gmfs/doppler.py (hunk 3 above)
Comment (ACHMartin) [reply seastar/gmfs/doppler.py:83]: Indeed, I moved it to tools. Thanks

**Pull request merged**